src/prover_disk.hpp

```diff
@@ -80,16 +80,14 @@ class DiskProver {
         } else {
             throw std::invalid_argument("Invalid plot file format");
         }
-
-        memcpy(this->id, header.id, sizeof(header.id));
+        memcpy(id.data(), header.id, sizeof(header.id));
         this->k = header.k;
         SafeSeek(disk_file, offsetof(struct plot_header, fmt_desc) + fmt_desc_len);
 
         uint8_t size_buf[2];
         SafeRead(disk_file, size_buf, 2);
-        this->memo_size = Util::TwoBytesToInt(size_buf);
-        this->memo = new uint8_t[this->memo_size];
-        SafeRead(disk_file, this->memo, this->memo_size);
+        memo.resize(Util::TwoBytesToInt(size_buf));
```

Comment:
this function should have been called `TwoBytesToUint()`. it does return an unsigned int so this is safe, but I had to look it up.

Reply:
Wasn't me! :smile: